userservice/src/main/java/ro/devdepot/services/implementation/UserServiceImpl.java

```diff
@@ -27,4 +40,106 @@ public LoginResponse login(LoginRequest loginRequest) {
 
         return null;
     }
+
+    @Override
+    public String createUser(CreateUserRequest createUserRequest) {
+
+        validateUserDto(createUserRequest);
+
+        User userCreated = userRepository.save(userMapper.getUserFrom(createUserRequest));
+
+        if(userCreated.getId() != null) {
+            return "User created with successfully";
+        }
+        throw new BusinessException("Something goes wrong", HttpStatus.CONFLICT);
+    }
+
+    @Override
+    public String updateUser(UpdateUserRequest updateUserRequest, Long id) {
+
+        User user = userRepository.findById(id).orElseThrow(() -> new BusinessException("User with id: " + id + " was not found", HttpStatus.NOT_FOUND));
+        user.setEmail(updateUserRequest.getEmail());
+        user.setFirstName(updateUserRequest.getFirstName());
+        user.setLastName(updateUserRequest.getLastName());
+        userRepository.save(user);
+
+        return "User " + user.getUsername() + " has been saved successfully";
+    }
+
+    @Override
+    public String updateUserRole(String userrole, Long id) {
+
+        User user = userRepository.findById(id).orElseThrow(() -> new BusinessException("User with id: " + id + " was not found", HttpStatus.NOT_FOUND));
+        try {
+            UserRole role = UserRole.valueOf(userrole.toUpperCase());
+
+            if(role == user.getUserRole()){
+                throw new BusinessException("Your user role does not have to be the same as your current user role", HttpStatus.BAD_REQUEST);
```

Comment:
why? if you update from a role to the same role, just do nothing, there is no reason to notify the user